src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { Person } from '../../types/Person';
+import { PersonList } from '../PersonList';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[],
+  onSelect: (person: Person) => void,
+  delay: number,
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelect,
+  delay,
+}) => {
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [debauncedInputValue, setDebauncedInputValue] = useState(inputValue);
+  const [wasThereClick, setWasThereClick] = useState(true);
+
+  const handleInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (event.target.value === '') {
+      setIsListVisible(true);
+      setWasThereClick(false);
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setIsListVisible(false);
+    setWasThereClick(false);
+  };
+
+  const handleItemClick = useCallback((person: Person) => {
+    onSelect(person);
+    setInputValue(person.name);
+    setIsListVisible(false);
+    setWasThereClick(true);
+  }, []);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      if (inputValue !== debauncedInputValue) {
+        setDebauncedInputValue(inputValue);
+      }
+
+      setIsListVisible(true);
+    }, delay);
+
+    return () => clearTimeout(timerId);
+  }, [inputValue]);
```

Comment:
The useEffect hook is missing dependencies in its array. This could lead to outdated data being used inside the hook. Ensure that all variables used inside the hook are included in the dependency array, in this case 'delay' and 'debauncedInputValue' should be included.